src/components/tl/ucp/tl_ucp_ep.c

```diff
@@ -0,0 +1,74 @@
+#include "tl_ucp.h"
+#include "tl_ucp_ep.h"
+
+ucc_status_t ucc_tl_ucp_connect_ep(ucc_tl_ucp_context_t *ctx,
+                                   ucc_tl_ucp_team_t *team, char *addr_array,
+                                   size_t max_addrlen, int rank)
+{
+    ucp_address_t  *address = (ucp_address_t *)(addr_array + max_addrlen * rank);
+    ucp_ep_params_t ep_params;
+    ucs_status_t    status;
+    ucp_ep_h       *ep;
+    if (team->context_ep_storage) {
```

Comment:
FYI, this would be applicable only in the case of shared context, and you might want to check it.

Reply:
This is correct. I have this as TODO at line 20 in tl_ucp_team.c. There i initialize context_ep_storage either to 1 or 0. The thing is : the check is more complicated then just context being shared or not. In order to user context ep storage we need the context to be "collective" (ie oob provided), we need the context to have defined EP with EP_RANGE_CONTIG (ie context level rank, or kind of world_rank) and we need the mappers from team_rank->context_rank provided at the team creation time. I plan to add this logic later because i think we might need some more work there (need to check all the cases when different mappings are provided and when not, and how to handle latter cases)